src/Interpreters/BlobStorageLog.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <Interpreters/SystemLog.h>
+#include <Core/NamesAndTypes.h>
+#include <Core/NamesAndAliases.h>
+#include <Poco/Message.h>
+#include <chrono>
+
+namespace DB
+{
+
+struct BlobStorageLogElement
+{
+    enum class EventType : Int8
+    {
+        Upload = 1,
+        Delete = 2,
+        MultiPartUploadCreate = 3,
+        MultiPartUploadWrite = 4,
+        MultiPartUploadComplete = 5,
+        MultiPartUploadAbort = 6,
+    };
+
+    EventType event_type;
+
+    String query_id;
+    UInt64 thread_id = 0;
+
+    String disk_name;
+    String bucket;
+    String remote_path;
+    String local_path;
+
+    UInt32 data_size;
+
+    Int32 error_code = -1; /// negative if no error
```

Comment:
Usually no error is code 0

Reply:
AWS SDK uses 0 as error code https://github.com/ClickHouse/aws-sdk-cpp/blob/6e9460f1b0a9af2aa7fd712fdefcc22e2aca6f66/src/aws-cpp-sdk-core/include/aws/core/client/CoreErrors.h#L21-L25